localstack-core/localstack/services/events/target.py

```diff
@@ -181,25 +181,20 @@ def _validate_input(self, target: Target):
             self._validate_input_transformer(input_transformer)
 
     def _initialize_client(self) -> BaseClient:
-        """Initializes internal boto client.
-        If a role from a target is provided, the client will be initialized with the assumed role.
-        If no role is provided, the client will be initialized with the account ID and region.
-        In both cases event bridge is requested as service principal"""
+        """Initializes internal boto client."""
         service_principal = ServicePrincipal.events
         role_arn = self.target.get("RoleArn")
-        if role_arn:  # required for cross account
-            # assumed role sessions expire after 6 hours in AWS, currently no expiration in LocalStack
```

Comment:
we should leave this comment hin since it is something that is currently not parity with AWS and could lead to problems (very unlikely but might be)